src/delaunator.cpp

```diff
@@ -442,7 +443,7 @@ long int Delaunator::hash_key(double x, double y) {
     // use pseudo-angle: a measure that monotonically increases
     // with real angle, but doesn't require expensive trigonometry
     const double p = 1 - dx / (abs(dx) + abs(dy));
-    return floor((2 + (dy < 0 ? -p : p)) / 4 * m_hash_size);
+    return floor(((2 + (dy < 0 ? -p : p)) / 4) * (m_hash_size-1));
```

Comment:
funny enough @mourner fixing (probably) the same or similar issue https://github.com/mapbox/delaunator/commit/d7a7ce61389cb27e4290769f3ab76255fa884e1c

I have a strong preference for consistency with JS version, as this project can benefit from JS version improvements.

Reply:
my assumption here was that this bit `((2 + (dy < 0 ? -p : p)) / 4)` is never bigger than 1, but i am not sure, it may not be the case. either way, yes the js fix makes sure it can never overflow in all cases and keeps compatibility with triangle indexes (unless where it overflows)